File: lambdas/saveUserToDynamo.js
// Lambda: saveUserToDynamo.js
const AWS = require('aws-sdk');
const dynamo = new AWS.DynamoDB.DocumentClient();

exports.handler = async (event) => {
  let body;
  try {
    body = typeof event.body === 'string' ? JSON.parse(event.body) : event.body;
  } catch {
    body = event;
  }
  const { username, password, fullname } = body;
  if (!username || !password || !fullname) {
    return { statusCode: 400, body: JSON.stringify({ error: 'Missing fields' }) };
  }
  const params = {
    TableName: 'ChargerGPT-Users',
    Item: { username, password, fullname }
  };
  try {
    await dynamo.put(params).promise();
    return { statusCode: 200, body: JSON.stringify({ success: true }) };
  } catch (err) {
    return { statusCode: 500, body: JSON.stringify({ error: err.message }) };
  }
};

File: lambdas/saveUserToDynamo.py
import json
import boto3

dynamo = boto3.resource('dynamodb')
table = dynamo.Table('ChargerGPT-Users')

def lambda_handler(event, context):
    # Try to parse body if it's a POST request
    if isinstance(event, dict) and "body" in event:
        try:
            body = json.loads(event["body"]) if isinstance(event["body"], str) else event["body"]
        except Exception:
            body = event
    else:
        body = event

    username = body.get("username")
    password = body.get("password")
    fullname = body.get("fullname")

    # Validate fields
    if not username or not password or not fullname:
        return {
            "statusCode": 400,
            "body": json.dumps({"error": "Missing fields"})
        }

    # Prepare DynamoDB item
    try:
        table.put_item(
            Item={
                "username": username,
                "password": password,
                "fullname": fullname
            }
        )
        return {
            "statusCode": 200,
            "body": json.dumps({"success": True})
        }

    except Exception as e:
        return {
            "statusCode": 500,
            "body": json.dumps({"error": str(e)})
        }
